app/Helpers/snmp.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * @param string $stat snmpget, snmpwalk
+ * @param float $start_time The time the operation started with 'microtime(true)'
+ * @return float  The calculated run time
+ */
+function recordSnmpStatistic($stat, $start_time)
+{
+    global $snmp_stats;
+    initStats();
+
+    $runtime = microtime(true) - $start_time;
+    $snmp_stats['ops'][$stat]++;
+    $snmp_stats['time'][$stat] += $runtime;
+    return $runtime;
+}
+
+/**
+ * Generate an snmpget command
+ *
+ * @param array $device the we will be connecting to
+ * @param string $oids the oids to fetch, separated by spaces
+ * @param string $options extra snmp command options, usually this is output options
+ * @param string $mib an additional mib to add to this command
+ * @param string $mibdir a mib directory to search for mibs, usually prepended with +
+ * @return string the fully assembled command, ready to run
+ */
+function gen_snmpget_cmd($device, $oids, $options = null, $mib = null, $mibdir = null)
+{
+    $snmpcmd  = Config::get('snmpget');
+    return gen_snmp_cmd($snmpcmd, $device, $oids, $options, $mib, $mibdir);
+} // end gen_snmpget_cmd()
+
+/**
+ * Generate an snmp command
+ *
+ * @param string $cmd the snmp command to run, like snmpget
+ * @param array $device the we will be connecting to
+ * @param string $oids the oids to fetch, separated by spaces
+ * @param string $options extra snmp command options, usually this is output options
+ * @param string $mib an additional mib to add to this command
+ * @param string $mibdir a mib directory to search for mibs, usually prepended with +
+ * @return string the fully assembled command, ready to run
+ */
+function gen_snmp_cmd($cmd, $device, $oids, $options = null, $mib = null, $mibdir = null)
+{
+    global $debug;
```

Comment:
@murrant I'm trying to think of a good way to migrate these global variables. Do you have any thoughts?